### PR TITLE
Jon readme

### PR DESCRIPTION
**Pull request by lejon1220**

### CODE REVIEW

**Comment by lejon1220** [comment]
Merged what is currently on dev with Kat's backend.
Added a jaeger file to the yaml folder.
Bug fixes associated with merging the new backend and front end.
Getall route now takes in and returns queries for the particular service

File: client/src/components/trace-table/tableListSlice.tsx [hunk 1]
```diff
@@ -65,8 +65,8 @@ export const getTraceTableDataAsync = createAsyncThunk(
 		}
 
 		console.log ('this is what the actual endpoint looks like', config.url + `/api/traces/getAll/${parameters.activeService}/${parameters.lookback}`)
-   // const url = config.url + `/api/traces/getAll/${parameters.activeService}/${parameters.lookback}`
-	 const url = config.url + `/api/traces/getAll/${parameters.lookback}`
+    const url = config.url + `/api/traces/getAll/${parameters.activeService}/${parameters.lookback}`
+	 //const url = config.url + `/api/traces/getAll/${parameters.lookback}`
```
Comment (drewdunne) [inline client/src/components/trace-table/tableListSlice.tsx]: can probably toss this log right?

File: server/controllers/TraceController.ts [hunk 2]
```diff
@@ -21,52 +21,84 @@ export class TraceController {
         });
       }
     }
-
-    public static getTraceViewData(req: Request, res: Response, next: NextFunction) {
+    public static getSearchBarTraceView(req: Request, res: Response, next: NextFunction) {
       try{
         console.log("In Tracecontroller");
-        TraceModel.getIndividualTraceView(req, res, next);
+        TraceModel.getSearchBarTraceView(req, res, next);
```
Comment (drewdunne) [inline server/controllers/TraceController.ts:27]: Should this function be named getTraceViewData or something like that? I also gets data when a trace is clicked on in the Trace Table does it not?

File: server/controllers/TraceController.ts [hunk 3]
```diff
@@ -21,52 +21,84 @@ export class TraceController {
...
       }
       catch(err) {
         return next({
           log: `Error in TraceController.getCluster: ${err}`,
           status: 500,
           message: {
-            err: 'Error occured while retrieving Trace View Data in TraceController',
+            err: 'Error occured while retrieving search bar trace data in TraceController',
```
Comment (drewdunne) [inline server/controllers/TraceController.ts:34]: is specifically calling out search bar here

File: server/controllers/TraceController.ts [hunk 4]
```diff
@@ -21,52 +21,84 @@ export class TraceController {
...
+    //   catch(err) {
+    //     return next({
+    //       log: `Error in TraceController.getCluster: ${err}`,
+    //       status: 500,
+    //       message: {
+    //         err: 'Error occured while retrieving Span Details Data in TraceController',
+    //       },
+    //     });
+    //   }
+    // }
```
Comment (drewdunne) [inline server/controllers/TraceController.ts:101]: A lot of commented out code here. Should we be keeping this in the codebase?

File: server/controllers/TraceController.ts [hunk 5]
```diff
@@ -21,52 +21,84 @@ export class TraceController {
...
     // public static saveData (req: Request, res: Response, next: NextFunction) {
     //       TraceModel.saveDataToTextFile(req, res);
     //       next();
```
Comment (drewdunne) [inline server/controllers/TraceController.ts:104]: Same goes for the stuff down here.

File: server/routers/TraceRouter.ts [hunk 6]
```diff
@@ -19,45 +19,35 @@ router.get('/getTraceView/:traceId',
...
-}
-);
+router.get('/getTraceViewServices', 
+  TraceController.getTraceViewServices,
+  (req: Request, res: Response, next: NextFunction) => {
+    console.log('router complete getAll');
+    res.status(200).json(res.locals.traceViewServices);
+  }
+)
+// Will need to pass in req.body the specific pod name, req.body.traceID
```
Comment (drewdunne) [inline server/routers/TraceRouter.ts:36]: These are get requests so I'm not certain this comment is relevant or even possible anymore unless we intend to send POSTs

File: server/routers/TraceRouter.ts [hunk 7]
```diff
@@ -19,45 +19,35 @@ router.get('/getTraceView/:traceId',
...
 //   TraceController.getSpanDetails,
 //   (req: Request, res: Response, next: NextFunction) => {
 //     res.status(200).json(res.locals.spanDetails)
-// } 
+//   } 
 // )
+
 // router.post('/post', 
 //   TraceController.saveData, 
 //   (req: Request, res: Response, next: NextFunction) => {res.status(200).json("Trace Data Added")}
```
Comment (drewdunne) [inline server/routers/TraceRouter.ts:53]: Lines 44-53 are all commented out. Can this go?

File: server/types/TraceData.ts [hunk 8]
```diff
@@ -4,4 +4,8 @@ type TraceData = {
     label: string,
     type: string
    }
-}
\ No newline at end of file
+}
+
+
+
+// pod name;
```
Comment (drewdunne) [inline server/types/TraceData.ts:11]: Seems like a stray comment

File: server/types/Types.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+export type EdgeData = {
+    data: {
+        source: string,
+        target: string,
+        type: 'arrow',
+        label: any,
+    },
+        classes: 'background'
+   };
```
Comment (drewdunne) [inline server/types/Types.ts:15]: This looks poorly linted. We need to install eslint to get things cleaned up at some point. Let's add a task for it?

File: server/controllers/TraceController.ts (hunk 2 above)
Comment (lejon1220) [reply server/controllers/TraceController.ts:27]: No, this populates the search bar with the relevant info.

File: server/controllers/TraceController.ts (hunk 4 above)
Comment (lejon1220) [reply server/controllers/TraceController.ts:101]: we need to keepthis forrichard.

**Pull request merged**